### PR TITLE
Moved deployment note to api migration pages and added XML JSON transformation information.

### PR DESCRIPTION
**Pull request by prasa7**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-200-to-400.md [hunk 1]
```diff
@@ -3526,6 +3548,52 @@ This concludes the upgrade process.
     The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse artifacts, execution plans, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
 
 !!! note
-    If you are using a migrated API and need to consume it via an application that supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. JWT authentication will not work if you do not republish the API, as it looks for a local entry that will get populated while publishing.
-
+   - If you are using a migrated API and need to consume it via an application that supports JWT authentication (default type since API-M 3.2.0), you need to republish the API. JWT authentication will not work if you do not republish the API, as it looks for a local entry that will get populated while publishing.
     You can consume the migrated API via an OAuth2 application without an issue.
+    
+   - API-M 4.0.0 synapse artifacts have been removed from the file system and are managed via database. At server startup the synapse configs are loaded to the memory from the Traffic Manager.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-200-to-400.md:3554]:
```suggestion
   - API-M 4.0.0 Synapse artifacts have been removed from the file system and are managed via database. At server startup the Synapse configs are loaded to the memory from the Traffic Manager.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-200-to-400.md [hunk 2]
```diff
@@ -3526,6 +3548,52 @@ This concludes the upgrade process.
...
+  
+   - When Migrating a Kubernetes environment to a newer API Manager version it is recommended to do the data migration in a single container and then do the deployment.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-200-to-400.md:3556]:
```suggestion
   - When migrating a Kubernetes environment to a newer API Manager version, it is recommended to do the data migration in a single container and then do the deployment.    
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-210-to-400.md [hunk 3]
```diff
@@ -3437,7 +3457,51 @@ This concludes the upgrade process.
     The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse artifacts, execution plans, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
 
 !!! note
-    If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
-
+   - If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
     You can consume the migrated API via an OAuth2 application without an issue.
+    
+   - API-M 4.0.0 synapse artifacts have been removed from the file system and are managed via database. At server startup the synapse configs are loaded to the memory from the Traffic Manager.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-210-to-400.md:3463]:
```suggestion
   - API-M 4.0.0 Synapse artifacts have been removed from the file system and are managed via database. At server startup the Synapse configs are loaded to the memory from the Traffic Manager.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-210-to-400.md [hunk 4]
```diff
@@ -3437,7 +3457,51 @@ This concludes the upgrade process.
...
+  
+   - When Migrating a Kubernetes environment to a newer API Manager version it is recommended to do the data migration in a single container and then do the deployment.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-210-to-400.md:3465]:
```suggestion
   - When Migrating a Kubernetes environment to a newer API Manager version, it is recommended to do the data migration in a single container and then do the deployment.    
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-220-to-400.md [hunk 5]
```diff
@@ -3002,7 +3010,21 @@ Follow the instructions below to move all the existing API Manager configuration
...
 
-6.  Upgrade the Identity component in WSO2 API Manager from version 5.5.0 to 5.11.0.
+6.  Configure the [SymmetricKeyInternalCryptoProvider](https://is.docs.wso2.com/en/5.11.0/administer/symmetric-overview/) as the default internal cryptor provider.
+    Generate your own secret key using a tool like OpenSSL.
+    
+    i.e.,
+       ```
+        openssl enc -nosalt -aes-128-cbc -k hello-world -P
+       ```        
+    Add the configuration to the <NEW_IS_HOME>/repository/conf/deployment.toml file.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-220-to-400.md:3020]:
```suggestion
    Add the configuration to the `<NEW_IS_HOME>/repository/conf/deployment.toml` file.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-220-to-400.md [hunk 6]
```diff
@@ -3225,6 +3247,51 @@ This concludes the upgrade process.
     The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse artifacts, execution plans, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
 
 !!! note
-    If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
-
+   - If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
     You can consume the migrated API via an OAuth2 application without an issue.
+    
+   - API-M 4.0.0 synapse artifacts have been removed from the file system and are managed via database. At server startup the synapse configs are loaded to the memory from the Traffic Manager.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-220-to-400.md:3253]:
```suggestion
   - API-M 4.0.0 Synapse artifacts have been removed from the file system and are managed via database. At server startup the Synapse configs are loaded to the memory from the Traffic Manager.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-220-to-400.md [hunk 7]
```diff
@@ -3225,6 +3247,51 @@ This concludes the upgrade process.
...
+   
+   - When Migrating a Kubernetes environment to a newer API Manager version it is recommended to do the data migration in a single container and then do the deployment.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-220-to-400.md:3255]:
```suggestion
   - When Migrating a Kubernetes environment to a newer API Manager version, it is recommended to do the data migration in a single container and then do the deployment.    
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-400.md [hunk 8]
```diff
@@ -2921,7 +2928,21 @@ Follow the instructions below to move all the existing API Manager configuration
...
 
-5.  Upgrade the Identity component in WSO2 API Manager from version 5.6.0 to 5.11.0.
+5.  Configure the [SymmetricKeyInternalCryptoProvider](https://is.docs.wso2.com/en/5.11.0/administer/symmetric-overview/) as the default internal cryptor provider.
+    Generate your own secret key using a tool like OpenSSL.
+    
+    i.e.,
+       ```
+        openssl enc -nosalt -aes-128-cbc -k hello-world -P
+       ```        
+    Add the configuration to the <NEW_IS_HOME>/repository/conf/deployment.toml file.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-400.md:2938]:
```suggestion
    Add the configuration to the `<NEW_IS_HOME>/repository/conf/deployment.toml` file.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-400.md [hunk 9]
```diff
@@ -3146,6 +3167,51 @@ This concludes the upgrade process.
         The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse artifacts, execution plans, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
 
 !!! note
-    If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
-
+   - If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
     You can consume the migrated API via an OAuth2 application without an issue.
+    
+   - API-M 4.0.0 synapse artifacts have been removed from the file system and are managed via database. At server startup the synapse configs are loaded to the memory from the Traffic Manager.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-400.md:3173]:
```suggestion
   - API-M 4.0.0 Synapse artifacts have been removed from the file system and are managed via database. At server startup the Synapse configs are loaded to the memory from the Traffic Manager.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-400.md [hunk 10]
```diff
@@ -3146,6 +3167,51 @@ This concludes the upgrade process.
...
+
+   - When Migrating a Kubernetes environment to a newer API Manager version it is recommended to do the data migration in a single container and then do the deployment.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-250-to-400.md:3175]:
```suggestion
   - When Migrating a Kubernetes environment to a newer API Manager version, it is recommended to do the data migration in a single container and then do the deployment.    
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-260-to-400.md [hunk 11]
```diff
@@ -3499,6 +3522,51 @@ This concludes the upgrade process.
     The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse artifacts, execution plans, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
 
 !!! note
-    If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
-
+   - If you are using a migrated API and wants to consume it via an application which supports JWT authentication (default type in API-M 4.0.0), you need to republish the API. Without republishing the API, JWT authentication doesn't work as it looks for a local entry which will get populated while publishing.
     You can consume the migrated API via an OAuth2 application without an issue.
+    
+   - API-M 4.0.0 synapse artifacts have been removed from the file system and are managed via database. At server startup the synapse configs are loaded to the memory from the Traffic Manager.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-260-to-400.md:3528]:
```suggestion
   - API-M 4.0.0 Synapse artifacts have been removed from the file system and are managed via database. At server startup the Synapse configs are loaded to the memory from the Traffic Manager.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-260-to-400.md [hunk 12]
```diff
@@ -3499,6 +3522,51 @@ This concludes the upgrade process.
...
+    
+   - When Migrating a Kubernetes environment to a newer API Manager version it is recommended to do the data migration in a single container and then do the deployment.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-260-to-400.md:3530]:
```suggestion
   - When Migrating a Kubernetes environment to a newer API Manager version, it is recommended to do the data migration in a single container and then do the deployment.    
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-310-to-400.md [hunk 13]
```diff
@@ -2251,3 +2270,50 @@ This concludes the upgrade process.
 
 !!! tip
     The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse configurations, execution plans and etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
+    
+!!! note
+   - API-M 4.0.0 synapse artifacts have been removed from the file system and are managed via database. At server startup the synapse configs are loaded to the memory from the Traffic Manager.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-310-to-400.md:2275]:
```suggestion
   - API-M 4.0.0 Synapse artifacts have been removed from the file system and are managed via database. At server startup the Synapse configs are loaded to the memory from the Traffic Manager.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-310-to-400.md [hunk 14]
```diff
@@ -2251,3 +2270,50 @@ This concludes the upgrade process.
...
+
+   - When Migrating a Kubernetes environment to a newer API Manager version it is recommended to do the data migration in a single container and then do the deployment.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-310-to-400.md:2277]:
```suggestion
   - When Migrating a Kubernetes environment to a newer API Manager version, it is recommended to do the data migration in a single container and then do the deployment.    
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-320-to-400.md [hunk 15]
```diff
@@ -1393,7 +1399,22 @@ Follow the instructions below to move all the existing API Manager configuration
...
-5.  Upgrade the Identity component in WSO2 API Manager from version 5.10.0 to 5.11.0.
+
+6.  Configure the [SymmetricKeyInternalCryptoProvider](https://is.docs.wso2.com/en/5.11.0/administer/symmetric-overview/) as the default internal cryptor provider.
+    Generate your own secret key using a tool like OpenSSL.
+    
+    i.e.,
+       ```
+        openssl enc -nosalt -aes-128-cbc -k hello-world -P
+       ```        
+    Add the configuration to the <NEW_IS_HOME>/repository/conf/deployment.toml file.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-320-to-400.md:1410]:
```suggestion
    Add the configuration to the `<NEW_IS_HOME>/repository/conf/deployment.toml` file.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-320-to-400.md [hunk 16]
```diff
@@ -1564,3 +1585,49 @@ This concludes the upgrade process.
 
 !!! tip
     The migration client that you use in this guide automatically migrates your tenants, workflows, external user stores, synapse artifacts, execution plans, etc. to the upgraded environment. Therefore, there is no need to migrate them manually.
+!!! note
+   - API-M 4.0.0 synapse artifacts have been removed from the file system and are managed via database. At server startup the synapse configs are loaded to the memory from the Traffic Manager.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-320-to-400.md:1589]:
```suggestion
   - API-M 4.0.0 Synapse artifacts have been removed from the file system and are managed via database. At server startup the Synapse configs are loaded to the memory from the Traffic Manager.
```

File: en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-320-to-400.md [hunk 17]
```diff
@@ -1564,3 +1585,49 @@ This concludes the upgrade process.
...
+
+   - When Migrating a Kubernetes environment to a newer API Manager version it is recommended to do the data migration in a single container and then do the deployment.
```
Comment (Mariangela) [inline en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-320-to-400.md:1591]:
```suggestion
   - When Migrating a Kubernetes environment to a newer API Manager version, it is recommended to do the data migration in a single container and then do the deployment.    
```

**Review by Mariangela — APPROVED** [review]

**Comment by Mariangela** [comment]
@prasa7 , please resolve merge conflict.

**Comment by prasa7** [comment]
This PR is resent from a separate PR https://github.com/wso2/docs-apim/pull/3758

**Pull request closed without merging**